sympy/physics/mechanics/abstract_body.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import Symbol, sympify
+from sympy.physics.vector import Point
+
+__all__ = ['_Body']
+
+
+class _Body(ABC):
+    """Abstract class for body type objects."""
+    def __init__(self, name, masscenter=None, mass=None):
+        if not isinstance(name, str):
+            raise TypeError('Supply a valid name.')
+        self._name = name
+        if mass is None:
+            mass = Symbol(f'{name}_mass')
+        if masscenter is None:
+            masscenter = Point(f'{name}_masscenter')
+        self.mass = mass
+        self.masscenter = masscenter
+        self.potential_energy = 0
+        self.points = []
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        return self.__str__()
```

Comment:
These should be improved, but again I assume they are currently like this to meet backwards compatibility?

Reply:
This is the backward compatible variant, but in case of the `__repr__` I'm personally fine with making it a bit better. Do note that it is not possible to exactly create a string, which would replicate the body, as frame and masscenter are both part of graphs.

Reply:
I think even if we can't make `__repr__` produce a string to directly recreate an instance, we can still make it much more informative than it currently is.

Reply:
If only `__repr__` is provided then `str(obj)` will use it so there shouldn't be a need to have one method call the other.

Reply:
Implemented an improved `__repr__` method for all body objects.